internal/pubsub/example_test.go

```diff
@@ -101,3 +103,147 @@ func Example_sendReceiveMultipleMessages() {
 	// b
 	// c
 }
+
+func Example_receiveWithInvertedWorkerPool() {
+	// Open a topic and corresponding subscription.
+	ctx := context.Background()
+	b := mempubsub.NewBroker([]string{"myTopic"})
+	t := mempubsub.OpenTopic(b, "myTopic")
+	defer t.Close()
+	s := mempubsub.OpenSubscription(b, "myTopic", time.Second)
+	defer s.Close()
+
+	// Send a bunch of messages to the topic.
+	const nMessages = 100
+	for n := 0; n < nMessages; n++ {
+		m := &pubsub.Message{
+			Body: []byte(fmt.Sprintf("message %d", n)),
+		}
+		if err := t.Send(ctx, m); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	// In order to make our test exit, we keep track of how many messages were
+	// processed and cancel the receiveCtx when we've processed them all.
+	var nProcessed int32
+	receiveCtx, cancel := context.WithCancel(ctx)
+	go func() {
+		for {
```

Comment:
This burns CPU.

You could use a waitgroup instead:
- Add(100) before publishing.
- Call Done when finished processing.
- Call Wait instead of this loop.